mirgecom/viscous.py

```diff
@@ -0,0 +1,402 @@
+        The discretization to use
+    eos: :class:`~mirgecom.eos.GasEOS`
+        A gas equation of state with a non-empty
+        :class:`~mirgecom.transport.TransportModel`
+    cv: :class:`~mirgecom.fluid.ConservedVars`
+        Fluid state
+    grad_cv: :class:`~mirgecom.fluid.ConservedVars`
+        Gradient of the fluid state
+
+    Returns
+    -------
+    numpy.ndarray
+        The species diffusive flux vector, $\mathbf{J}_{\alpha}$
+    """
+    transport = eos.transport_model()
+
+    grad_y = species_mass_fraction_gradient(discr, cv, grad_cv)
+    d = transport.species_diffusivity(eos, cv)
+    return -cv.mass*d.reshape(-1, 1)*grad_y
+
+
+def conductive_heat_flux(discr, eos, cv, grad_t):
+    r"""Compute the conductive heat flux, ($\mathbf{q}_{c}$).
+
+    The conductive heat flux is defined by:
+
+    .. math::
+
+        \mathbf{q}_{c} = -\kappa\nabla{T},
+
+    with thermal conductivity $\kappa$, and gas temperature $T$.
+
+    Parameters
+    ----------
+    discr: :class:`grudge.eager.EagerDGDiscretization`
+        The discretization to use
+    eos: :class:`~mirgecom.eos.GasEOS`
+        A gas equation of state with a non-empty
+        :class:`~mirgecom.transport.TransportModel`
+    cv: :class:`~mirgecom.fluid.ConservedVars`
+        Fluid state
+    grad_t: numpy.ndarray
+        Gradient of the fluid temperature
+
+    Returns
+    -------
+    numpy.ndarray
+        The conductive heat flux vector
+    """
+    transport = eos.transport_model()
+    return -transport.thermal_conductivity(eos, cv)*grad_t
+
+
+def diffusive_heat_flux(discr, eos, cv, j):
+    r"""Compute the diffusive heat flux, ($\mathbf{q}_{d}$).
+
+    The diffusive heat flux is defined by:
+
+    .. math::
+
+        \mathbf{q}_{d} = \sum_{\alpha=1}^{\mathtt{Nspecies}}{h}_{\alpha}
+        \mathbf{J}_{\alpha},
+
+    with species specific enthalpy ${h}_{\alpha}$ and diffusive flux
+    ($\mathbf{J}_{\alpha}$) defined as:
+
+    .. math::
+
+        \mathbf{J}_{\alpha} = -\rho{d}_{\alpha}\nabla{Y}_{\alpha},
+
+    where ${Y}_{\alpha}$ is the vector of species mass fractions.
+
+    Parameters
+    ----------
+    discr: :class:`grudge.eager.EagerDGDiscretization`
+        The discretization to use
+    eos: mirgecom.eos.GasEOS
+        A gas equation of state with a non-empty
+        :class:`~mirgecom.transport.TransportModel`
+    cv: :class:`~mirgecom.fluid.ConservedVars`
+        Fluid state
+    j: numpy.ndarray
+        The species diffusive flux vector
+
+    Returns
+    -------
+    numpy.ndarray
+        The total diffusive heat flux vector
+    """
+    if isinstance(eos, MixtureEOS):
+        h_alpha = eos.species_enthalpies(cv)
+        return sum(h_alpha.reshape(-1, 1) * j)
+    return 0
+
+
+# TODO: We could easily make this more general (dv, grad_dv)
+# so that user can define arbitrary array containers for *dv*,
+# enabling support for other models and simultaneous eqns.
+def viscous_flux(discr, eos, cv, grad_cv, grad_t):
```

Comment:
Is this comment really true? At least `species_diffusivity` seems to need to know "who it's talking to".

Reply:
> Is this comment really true? At least `species_diffusivity` seems to need to know "who it's talking to".

Isn't species diffusivity just a function of (CV, grad(CV))? My only thought here, and perhaps it is misguided, is that if we had a DV array container (like CV), we could use to send around 
DV, and the gradient of DV.  Regardless, I will remove the comment.